libraries/MySensors/examples/AirQualitySensor/AirQuality-CO2-MH-Z14.ino

```diff
@@ -0,0 +1,76 @@
+/*
+
+  MH-Z14 CO2 sensor
+
+  Wiring:
+    Pad 1, Pad 5: Vin (Voltage input 4.5V-6V) 
+    Pad 2, Pad 3, Pad 12: GND 
+    Pad 6: PWM output ==> pin 6
+
+	From: http://davidegironi.blogspot.fr/2014/01/co2-meter-using-ndir-infrared-mh-z14.html
+	  MH-Z14 has a PWM output, with a sensitivity range of 0ppm to 2000ppm CO2, an accurancy of ±200ppm.
+	  The cycle is 1004ms±5%, given the duty cicle Th (pulse high), Tl is 1004-Th, we can convert it to CO2 value using the formula:
+	  CO2ppm = 2000 * (Th - 2ms) /(Th + Tl - 4ms)
+	From: http://airqualityegg.wikispaces.com/Sensor+Tests
+	  - response time is less than 30 s
+    - 3 minute warm up time
+	datasheet: http://www.futurlec.com/Datasheet/Sensor/MH-Z14.pdf
+	Contributor: epierre
+*/
+
+
+#include <MySensor.h>  
+#include <SPI.h>
+
+#define CHILD_ID_AIQ 0
+#define AIQ_SENSOR_ANALOG_PIN 6
+
+float valAIQ =0.0;
+float lastAIQ =0.0;
+
+MySensor gw;
+MyMessage msg(CHILD_ID_AIQ, V_LEVEL);
+
```

Comment:
HEllo,

  is that a new message or part of the existing message ?

On Mon, Aug 10, 2015 at 8:50 PM, Henrik Ekblad <notifications@github.com>
wrote:

> In libraries/MySensors/examples/AirQualitySensor/AirQuality-CO2-MH-Z14.ino
> <https://github.com/mysensors/Arduino/pull/164#discussion_r36669364>:
>
> > +	Contributor: epierre
> > +*/
> > +
> > +
> > +#include <MySensor.h>
> > +#include <SPI.h>
> > +
> > +#define CHILD_ID_AIQ 0
> > +#define AIQ_SENSOR_ANALOG_PIN 6
> > +
> > +float valAIQ =0.0;
> > +float lastAIQ =0.0;
> > +
> > +MySensor gw;
> > +MyMessage msg(CHILD_ID_AIQ, V_LEVEL);
> > +
>
> Would be great if you also could send in an textual V_UNIT_PREFIX message
> containing the unit ("ppm" in this case) in setup().
>
> This is optional but would make things simpler when presenting the sensor
> data in controller.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mysensors/Arduino/pull/164/files#r36669364>.
>